src/Aspire.Hosting.Milvus/MilvusBuilderExtensions.cs

```diff
@@ -0,0 +1,101 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.Milvus;
+using Aspire.Hosting.Utils;
+
+namespace Aspire.Hosting;
+
+/// <summary>
+/// Provides extension methods for adding Milvus resources to the application model.
+/// </summary>
+public static class MilvusBuilderExtensions
+{
+    private const int MilvusPortGrpc = 19530;
+    //private const string DefaultApiKey = "root:Milvus";
+
+    /// <summary>
+    /// Adds a Milvus resource to the application. A container is used for local development.
+    /// </summary>
+    /// <remarks>
+    /// This version the package defaults to the v2.4.0 tag of the milvusdb/milvus container image.
+    /// The .NET client library uses the gRPC port by default to communicate and this resource exposes that endpoint.
+    /// </remarks>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/>.</param>
+    /// <param name="name">The name of the resource. This name will be used as the connection string name when referenced in a dependency</param>
+    /// <param name="apiKey">The parameter used to provide the auth key/token user for the Milvus resource.</param>
+    /// <param name="grpcPort">The host port of gRPC endpoint of Milvus database.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{MilvusServerResource}"/>.</returns>
+    public static IResourceBuilder<MilvusServerResource> AddMilvus(this IDistributedApplicationBuilder builder,
```

Comment:
Would be good to include `<example />` elements where. I'd probably have one example that shows with and without explicit auth keys. You might also `<see />` for `WithAttu` to and add an example of its wire-up there to make it more discoverable.

And yes we have to do this everywhere but trying to stop the bleeding :)